base/workerpool.jl

```diff
@@ -1,11 +1,22 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
-type WorkerPool
+abstract AbstractWorkerPool
+
+# An AbstractWorkerPool should implement
```

Comment:
and at least according to the way the methods below are using them, they should all have fields `workers` and `channel` ?